File: ts-sdk/src/layers/http.ts
import { BaseRecord, LogRecord } from '../core/record';
import { Layer } from '../core/enums';

export interface HttpRecordProps extends LogRecord {
  readonly method: string;
  readonly status_code: number;
  readonly latency_ms: number;
  readonly rate_limited: boolean;
}

export class HttpRecord extends BaseRecord implements HttpRecordProps {
  readonly method: string;
  readonly status_code: number;
  readonly latency_ms: number;
  readonly rate_limited: boolean;

  constructor(data: HttpRecordProps) {
    super(data);
    this.method = data.method || 'GET';
    this.status_code = data.status_code || 200;
    this.latency_ms = data.latency_ms || 0;
    this.rate_limited = !!data.rate_limited;
    Object.freeze(this);
  }

  toDict(): Record<string, any> {
    return { ...this, layer: Layer.GATEWAY };
  }
}

File: py-sdk/mylogger/layers/http.py
from dataclasses import dataclass
from typing import Optional, Dict, Any
from ..core.record import LogRecord, Layer, Level
from ..core.logger import Logger

@dataclass(frozen=True)
class HttpRecord(LogRecord):
    method: str = "GET"
    status_code: int = 200
    rate_limited: bool = False
    bot_score: Optional[float] = None

class GatewayLogger(Logger[HttpRecord]):
    def create_record(self, msg: str, level: Level, **ctx) -> HttpRecord:
        base = self._get_base_props(msg, level, Layer.GATEWAY, **ctx)
        return HttpRecord(
            **base,
            method=ctx.get("method", "GET"),
            status_code=ctx.get("status_code", 200),
            rate_limited=ctx.get("rate_limited", False),
            bot_score=ctx.get("bot_score")
        )

    def handle(self, record: HttpRecord) -> None:
        for handler in self.handlers:
            handler.handle(record)
